### PR TITLE
Support more system tasks/functions

### PR DESCRIPTION
**Pull request by kbieganski**
Not sure about the macros, but it does reduce boilerplate code.

Related test: https://github.com/chipsalliance/UHDM-integration-tests/pull/534

### CODE REVIEW

File: src/UhdmAst.cpp [hunk 1]
```diff
@@ -1703,6 +1703,21 @@ AstNode* process_function_task(vpiHandle obj_h, UhdmShared& shared) {
     return taskFuncp;
 }
 
+AstNode* vectorToAstList(std::vector<AstNode*>::const_iterator begin, std::vector<AstNode*>::const_iterator end) {
+    AstNode* node = nullptr;
```
Comment (RRozak) [inline src/UhdmAst.cpp]: Please change it to `nodep`, according to naming convention.

File: src/UhdmAst.cpp [hunk 2]
```diff
@@ -2538,215 +2553,231 @@ AstNode* visit_object(vpiHandle obj_h, UhdmShared& shared) {
...
+                    return new AstTime(make_fileline(obj_h), VTimescale::NONE);
+                })
+        };
+        auto it = sysFuncCallMakers.find(objectName);
+        if (it != sysFuncCallMakers.end())
+            node = it->second(obj_h, shared, arguments);
+        else
             v3error("\t! Encountered unhandled SysFuncCall: " << objectName);
+        if (VN_IS(node, NodeMath) && !VN_IS(node, Signed) && !VN_IS(node, Unsigned)) {
+            auto parent_h = vpi_handle(vpiParent, obj_h);
```
Comment (RRozak) [inline src/UhdmAst.cpp:2775]: This handle is never released.

**Review by RRozak — CHANGES_REQUESTED** [review]

File: src/UhdmAst.cpp (hunk 2 above)
Comment (kbieganski) [reply src/UhdmAst.cpp:2775]: Fixed

File: src/UhdmAst.cpp (hunk 1 above)
Comment (kbieganski) [reply src/UhdmAst.cpp]: Fixed, also used existing Verilator function for creating the list

File: src/UhdmAst.cpp (hunk 2 above)
Comment (RRozak) [reply src/UhdmAst.cpp:2775]: If `parent_type == vpiBegin` the handle still won't be released. Please move `vpi_release_handle(parent_h);` before that `if`.

**Review by RRozak — CHANGES_REQUESTED** [review]

File: src/UhdmAst.cpp (hunk 2 above)
Comment (kbieganski) [reply src/UhdmAst.cpp:2775]: Oops. Fixed.

**Review by RRozak — APPROVED** [review]

**Pull request merged**